composer/callbacks/optimizer_monitor.py

```diff
@@ -99,7 +99,7 @@ def batch_end(self, state: State, logger: Logger):
 
                 metric_reporter = getattr(state.optimizers[0], 'report_per_parameter_metrics', None)
                 if callable(metric_reporter) and self.log_optimizer_metrics:
-                    optimizer_metrics = metric_reporter(p, name, optimizer_metrics)
+                    optimizer_metrics.update(metric_reporter(p, name, optimizer_metrics))
```

Comment:
do we need to pass in `optimizer_metrics` as an arg if we're calling `update`?

Reply:
yes, the `metric_reporter` function has loops over `optimizer_metrics` and such.